### PR TITLE
For #5 – Bump OAuth League client

### PR DESCRIPTION
**Pull request by tinpansoul**
- Updated `league/oauth2-client` to stable 1.x version instead of beta 0.5
- Updated `guzzle/guzzle` 3.x (deprecated) to `guzzlehttp/guzzle` 6.x
- Add two VK API method-helpers: `users.get` and `friends.get`

### CODE REVIEW

**Comment by tinpansoul** [comment]
Here i need help with writing tests for getting access token and users/friends without real requests.

**Comment by tinpansoul** [comment]
`league/oauth2-client` requires PHP 5.5+ because of `guzzlehttp/guzzle` [this commit](https://github.com/thephpleague/oauth2-client/commit/e84bf29023ec96edc27e86795765b0e5bee1fbef).

**Comment by j4k** [comment]
Looks good - thanks for doing this!

File: src/User.php [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+     * @param string $key
+     *
+     * @return mixed|null
+     */
+    protected function getField($key)
+    {
+        return !empty($this->response[$key]) ? $this->response[$key] : null;
+    }
+
+    // ========== Getters for default scope ==========
```
Comment (j4k) [inline src/User.php]: These kinds of comments aren't really necessary and are a bit untidy - could you remove them? The method names are self explanatory

File: src/Vkontakte.php [hunk 2]
```diff
@@ -2,184 +2,238 @@
...
+
+    /**
+     * @type array
+     * @see https://vk.com/dev/permissions
+     */
+    public $scopes = [
+        'email',
+        'friends',
+        'offline',
+        //'photos',
```
Comment (j4k) [inline src/Vkontakte.php]: Don't leave these commented out like this - remove them if they aren't in use, and allow consumers to pass in additional scopes

File: src/Vkontakte.php [hunk 3]
```diff
@@ -2,184 +2,238 @@
...
+        'id',
+        'is_friend',
+        'last_name',
+        'maiden_name',
+        'nickname',
+        'photo_max',
+        'photo_max_orig',
+        'screen_name',
+        'sex',
+        //'about',
```
Comment (j4k) [inline src/Vkontakte.php]: Same here

File: tests/VkontakteTest.php [hunk 4]
```diff
@@ -2,95 +2,291 @@
...
-        $this->assertEquals('/access_token', $uri['path']);
+        static::assertEquals('/access_token', $uri['path']);
+    }
+    public function testScopes()
+    {
+        static::assertEquals($this->defaultScopes, $this->provider->scopes);
     }
 
+    /**
+     * @todo Do we need this? Provider::getAccessToken() not used in project.
```
Comment (j4k) [inline tests/VkontakteTest.php]: You can remove this I think?

**Comment by j4k** [comment]
Can you fix anything that phpcs psr2 spits out please @TrogWarZ ? Also rebase some of the duplicated commit messages out of the pull

File: src/User.php (hunk 1 above)
Comment (tinpansoul) [inline src/User.php]: @j4k, do you mean "not-dockblocks"? Sure, i think same. They're just meta-blocks that can be moved to trait-helpers, value-object-class, etc.

File: tests/VkontakteTest.php (hunk 4 above)
Comment (tinpansoul) [inline tests/VkontakteTest.php]: Yes, but we need don't forget to write tests for getting access_token and user

File: src/User.php (hunk 1 above)
Comment (j4k) [inline src/User.php]: No - these guys `// ========== Getters for default scope ==========`

Comment (tinpansoul) [inline src/User.php]: Done

File: tests/VkontakteTest.php (hunk 4 above)
Comment (tinpansoul) [inline tests/VkontakteTest.php]: Done

**Comment by tinpansoul** [comment]
Done, fixed coding standards with phpcs and psr-2, thank you.

**Comment by anton-kotik** [comment]
Is `"league/oauth2-client": "^1.4"` really necessary? For example official providers require `"league/oauth2-client": "~1.0"`.

**Comment by tinpansoul** [comment]
@27cm, i just run `composer require league/oauth2-client` and it get "newest stable version" of package and really don't know difference between any other older stable versions.

I don't have time right now to switch 1.0.x and test vk-client with it (maybe, this week or so). Furthermore, i don't see any reason not to use "latest stable version" of package.

Can you tell me use-case example of situation when we need exactly 1.0.x-branch? They have any broken changes? If it so, then i can switch back requirement in composer and try to test it.

**Pull request merged**